src/herder/TransactionQueue.cpp

```diff
@@ -60,8 +60,19 @@ canReplaceByFee(TransactionFrameBasePtr tx, TransactionFrameBasePtr oldTx)
     // FEE_MULTIPLIER * v2 does not overflow uint128_t because fees are bounded
     // by INT64_MAX, while number of operations and FEE_MULTIPLIER are small.
     auto v1 = bigMultiply(newFee, oldNumOps);
-    auto v2 = bigMultiply(oldFee, newNumOps);
-    return v1 >= TransactionQueue::FEE_MULTIPLIER * v2;
+    auto v2 = uint128_t(TransactionQueue::FEE_MULTIPLIER) *
+              bigMultiply(oldFee, newNumOps);
+    std::pair<bool, int64_t> result;
+    result.first = v1 >= v2;
+    if (!result.first)
+    {
+        // Add a fee recommendation for future attempts
+        // recommendation = FEE_MULTIPLIER * oldFeeRate * newNumOps
+        //                = FEE_MULTIPLIER * (oldFee / oldNumOps) * newNumOps
+        //                = (FEE_MULTIPLIER * oldFee * newNumOps) / oldNumOps
+        result.second = bigDivide(v2, oldNumOps, ROUND_UP);
```

Comment:
Can we maybe reorganize the code a bit differently:
only one code path cares about this computation, so we could move it to a function `computeMinFee` of its own that gets called by the caller at the top if it runs into the `txINSUFFICIENT_FEE` error (see my other commend in the `tx` endpoint, related).

Reason this came up during my review is that `bigDivide` can throw and I'd hate to crash the other code paths.